### ISSUE TITLE
[Bug]: export is not correct when using rspack to output a library in esm

### ISSUE THREAD

**Issue by psYifanChen** [web-infra-dev/rspack#6572]
### System Info

System:
  OS: Windows 10 10.0.19045
  CPU: (4) x64 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
  Memory: 4.73 GB / 16.00 GB
Binaries:
  Node: 20.12.1 - C:\Program Files\nodejs\node.EXE
  Yarn: 3.6.1 - C:\Program Files\nodejs\yarn.CMD
  npm: 10.5.0 - C:\Program Files\nodejs\npm.CMD
  pnpm: 9.1.1 - C:\Program Files\nodejs\pnpm.CMD
Browsers:
  Edge: Chromium (124.0.2478.105)
  Internet Explorer: 11.0.19041.4355
npmPackages:
  @rspack/cli: ^0.6.5 => 0.5.9
  @rspack/core: ^0.6.5 => 0.5.9

### Details

I'm working on a serverless project using aws lambda, and we use rspack to bundle handlers writing in typescirpt like the following.

```ts
// index.ts
export const handler = (event: Event) => {
  return {
    status: 200,
    body: JSON.strinify({ message: 'hello world' })
  }
}
```

Before deploying it to aws, we need to transplie it to .js file, and then aws is responsible for calling the function at runtime like `import('./index.js').then(m => m.handler(event))` or in a cjs way.  
I know it's not the same as the library usecase but they're similar aren't they?

It works well under v0.5.9 but no longer works after updating rspack to v0.6.5.
I try to figure out the reason and found the following diff in output files.

```js
// v0.5.9
var e = {
  163: function() {
    // ...
    hello: function() {}
  }
}
var t = r('163').hello
export { t as hello };

// v0.6.5
var e = {
  163: function() {
    // ...
    o: function() {}
  }
}
var t = r('163').hello
export { t as hello };
```

I think it's the reason and hope it helpful.

### Reproduce link

https://github.com/psYifanChen/rspack-lib

### Reproduce Steps

Running the following commands under v0.5.9 and >=0.6.0 respectively and ovserve the console output

- `pnpm run build`
- `node index.js`

**Comment by LingyuCoder** [comment]
Seems to be a bug of `optimization.mangleExports`, cc @JSerFeng

**Comment by psYifanChen** [comment]
@LingyuCoder 
Btw, I also tried to output files in cjs but failed. 
I can not get something like `module.exports = { t as hello };` but only an IIFE. 
Could you give me a example for how to configure rspack to output files in cjs in library mode?

**Comment by LingyuCoder** [comment]
> @LingyuCoder Btw, I also tried to output files in cjs but failed. I can not get something like `module.exports = { t as hello };` but only an IIFE. Could you give me an example for how to configure rspack to output files in cjs in library mode?

try to set `output.library.type = "commonjs2"` like:

```js
/** @type {import("@rspack/core").Configuration} */
module.exports = {
  output: {
    library: {
      type: "commonjs2"
    }
  },
};

```

**Comment by psYifanChen** [comment]
@LingyuCoder 
Thanks! I got what I want by setting `output.library.type` to `commonjs2` or `commonjs`.

But I found another diff which may be a bug.
When setting `output.library.type` to `commonjs-static`, there is a diff like this.

```js
// v0.5.9
var r = n('294');
(exports.handler = r.handler), Object.defineProperty(exports, '__esModule', { value: !0 });

// v0.6.5
n('294'),
Object.defineProperty(exports, '__esModule', { value: !0 });
```

**Comment by LingyuCoder** [comment]
> @LingyuCoder Thanks! I got what I want by setting `output.library.type` to `commonjs2` or `commonjs`.
> 
> But I found another diff which may be a bug. When setting `output.library.type` to `commonjs-static`, there is a diff like this.
> 
> ```js
> // v0.5.9
> var r = n('294');
> (exports.handler = r.handler), Object.defineProperty(exports, '__esModule', { value: !0 });
> 
> // v0.6.5
> n('294'),
> Object.defineProperty(exports, '__esModule', { value: !0 });
> ```
> 
> Just reporting.

cc @ahabhgk , relate to full aligning library bootstrap with webpack

**Comment by ahabhgk** [comment]
Yes, `output.library.type = 'commonjs-static'` should already fixed in the main branch, the fix will be released in v0.7.0

and @JSerFeng will fix the mangleExports soon, should also be released in v0.7.0
